src/internal/agent/hooks/flux-ocirepo.go

```diff
@@ -0,0 +1,134 @@
+// SPDX-License-Identifier: Apache-2.0
+// SPDX-FileCopyrightText: 2021-Present The Zarf Authors
+
+// Package hooks contains the mutation hooks for the Zarf agent.
+package hooks
+
+import (
+	"context"
+	"encoding/json"
+	"fmt"
+
+	"github.com/defenseunicorns/pkg/helpers"
+	"github.com/defenseunicorns/zarf/src/config"
+	"github.com/defenseunicorns/zarf/src/config/lang"
+	"github.com/defenseunicorns/zarf/src/internal/agent/operations"
+	"github.com/defenseunicorns/zarf/src/pkg/cluster"
+	"github.com/defenseunicorns/zarf/src/pkg/message"
+	"github.com/defenseunicorns/zarf/src/pkg/transform"
+	"github.com/defenseunicorns/zarf/src/types"
+	"github.com/fluxcd/pkg/apis/meta"
+	flux "github.com/fluxcd/source-controller/api/v1beta2"
+	v1 "k8s.io/api/admission/v1"
+)
+
+// NewOCIRepositoryMutationHook creates a new instance of the oci repo mutation hook.
+func NewOCIRepositoryMutationHook(ctx context.Context, cluster *cluster.Cluster) operations.Hook {
+	message.Debug("hooks.NewOCIRepositoryMutationHook()")
+	return operations.Hook{
+		Create: func(r *v1.AdmissionRequest) (*operations.Result, error) {
+			return mutateOCIRepo(ctx, r, cluster)
+		},
+		Update: func(r *v1.AdmissionRequest) (*operations.Result, error) {
+			return mutateOCIRepo(ctx, r, cluster)
+		},
+	}
+}
+
+// mutateOCIRepo mutates the oci repository url to point to the repository URL defined in the ZarfState.
+func mutateOCIRepo(ctx context.Context, r *v1.AdmissionRequest, cluster *cluster.Cluster) (result *operations.Result, err error) {
+	var (
+		zarfState *types.ZarfState
+	)
+
+	src := &flux.OCIRepository{}
+	if err = json.Unmarshal(r.Object.Raw, &src); err != nil {
+		return nil, fmt.Errorf(lang.ErrUnmarshal, err)
+	}
+
+	// If we have a semver we want to continue since we wil still have the upstream tag
+	// but should warn that we can't guarantee there won't be collisions
+	if src.Spec.Reference.SemVer != "" {
+		message.Warnf(lang.AgentWarnSemVerRef, src.Spec.Reference.SemVer)
+	}
+
+	if src.Labels != nil && src.Labels["zarf-agent"] == "patched" {
+		return &operations.Result{
+			Allowed:  true,
+			PatchOps: []operations.PatchOperation{},
+		}, nil
+	}
+
+	if zarfState, err = cluster.LoadZarfState(ctx); err != nil {
```

Comment:
This logic here makes things a bit confusing. Because the zarfState is defined earlier the value will be assigned if error is not returned. I would much rather split the function calling and if statement to separate lines to make this more readable.